plugins/inputs/turbostat/parser/parser.go

```diff
@@ -0,0 +1,232 @@
+//go:build linux && amd64
+
+package parser
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/influxdata/telegraf"
+)
+
+var abbreviations = map[string]string{
+	"%":    "percent",
+	"+":    "plus",
+	"-":    "minus",
+	"watt": "power",
+	"mhz":  "frequency",
+	"tmp":  "temperature",
+	"thr":  "throttle",
+	"avg":  "average",
+	"cor":  "core",
+	"bzy":  "busy",
+	"pkg":  "package",
+	"sys":  "system",
+	"unc":  "uncore",
+	"u":    "uncore",
+	"a":    "actual",
+	"j":    "energy",
+}
+
+var keepSymbols = map[rune]struct{}{
+	'%': {},
+	'+': {},
+	'-': {},
+}
+
+var knownTokens = []string{
+	"Watt",
+	"MHz",
+	"Tmp",
+	"Thr",
+	"GFX",
+	"SAM",
+}
+
+var tagNames = map[string]struct{}{
+	"package": {},
+	"die":     {},
+	"core":    {},
+	"cpu":     {},
+	"apic":    {},
+	"x2apic":  {},
+}
+
+type column struct {
+	name       string
+	isTag      bool
+	isIgnored  bool
+	multiplier float64
+}
+
+type tagMap = map[string]string
+type fieldMap = map[string]any
+
+// Reads metrics from a stream (such as Turbostat stdout) and adds them
+// to an accumulator. If an error is encountered, the function returns it
+// and stops further processing.
+func ProcessStream(r io.Reader, acc telegraf.Accumulator) error {
```

Comment:
Comment on exported "things" should have the following format `// ProcessStream ...`

Reply:
Done, thanks.